movies/presentation/src/main/java/com/example/movies/presentation/ui/movieDetail/MovieDetailFragment.kt

```diff
@@ -41,28 +42,44 @@ class MovieDetailFragment : BaseFragment() {
             setContent {
                 InitializeStateVariables()
                 reviewChangeStatesUi()
-                DetailScreen(movieDetail.value, isLoadingVisible.value)
+                DetailScreen(
+                    movieDetail.value, movieRecommendations.value, isLoadingVisible.value,
+                    onNavigateDetail = { dest -> findNavController().navigate(dest) },
+                )
             }
         }
     }
 
     @Composable
     private fun InitializeStateVariables() {
         movieDetail = remember { mutableStateOf(null) }
+        movieRecommendations = remember { mutableStateOf(null) }
         isLoadingVisible = remember { mutableStateOf(false) }
     }
 
     @SuppressLint("UnsafeRepeatOnLifecycleDetector", "CoroutineCreationDuringComposition")
     private fun reviewChangeStatesUi() {
         lifecycleScope.launch {
             repeatOnLifecycle(Lifecycle.State.STARTED) {
-                viewModel.uiState.collect { uiState ->
+                viewModel.movieDetailUiState.collect { uiState ->
                     when (uiState) {
                         is MovieDetailUiState.Init -> isLoadingVisible.value = false
                         is MovieDetailUiState.Loading -> isLoadingVisible.value = true
                         is MovieDetailUiState.GetDetailInformation -> setSuccessMovieDetail(uiState)
-                        is MovieDetailUiState.Error -> showError(uiState)
+                        is MovieDetailUiState.Error -> showError(uiState.message)
+                    }
+                }
+            }
+        }
+        lifecycleScope.launch {
+            viewModel.movieRecommendationUiState.collect { uiState ->
+                when (uiState) {
+                    is MovieRecommendationUiState.Init -> isLoadingVisible.value = false
+                    is MovieRecommendationUiState.Loading -> isLoadingVisible.value = true
+                    is MovieRecommendationUiState.GetMovieRecommendations -> {
```

Comment:
this line should be MovieRecommendationUiState.GetMovieRecommendations ->                      setSuccessMovieRecommendations(uiState)